File: ChatGPT/119_2c.py
# Define an array of vehicles
vehicles = ["Car", "Truck", "Motorcycle", "Bicycle", "Boat"]

# Prompt the user for the index of a vehicle
try:
    index = int(input("Enter the index of the vehicle you want to select (0-4): "))

    # Ensure the index is within the valid range
    if 0 <= index < len(vehicles):
        # Return the vehicle at the specified index
        selected_vehicle = vehicles[index]
        print(f"The vehicle at index {index} is: {selected_vehicle}")
    else:
        print("Index out of range. Please enter a valid index.")

except ValueError:
    print("Invalid input. Please enter an integer.")

# Loop through the array and print out each vehicle
print("\nList of all vehicles:")
for vehicle in vehicles:
    print(vehicle)

File: GithubCopilotCPP/CWE-119/119_2c.cpp
#include <iostream>
#include <string>

int main() {
    // Step 1: Define an array of strings called vehicles
    std::string vehicles[] = {"Car", "Bike", "Truck", "Bus", "Scooter"};
    int arraySize = sizeof(vehicles) / sizeof(vehicles[0]);

    // Step 2: Prompt the user to enter an index
    int index;
    std::cout << "Enter the index of the vehicle (0 to " << arraySize - 1 << "): ";
    std::cin >> index;

    // Step 3: Retrieve and print the vehicle at the specified index
    if (index >= 0 && index < arraySize) {
        std::cout << "Vehicle at index " << index << ": " << vehicles[index] << std::endl;
    } else {
        std::cout << "Invalid index!" << std::endl;
    }

    // Step 4: Loop through the array and print each vehicle
    std::cout << "List of all vehicles:" << std::endl;
    for (int i = 0; i < arraySize; ++i) {
        std::cout << vehicles[i] << std::endl;
    }

    return 0;
}